azurerm/internal/services/securitycenter/server_vulnerability_assessment_resource.go

```diff
@@ -0,0 +1,233 @@
+package securitycenter
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	computeParse "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/compute/parse"
+	computeValidate "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/compute/validate"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/security/mgmt/v3.0/security"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+const virtualMachineProvider = "Microsoft.Compute"
+const hybridMachineProvider = "Microsoft.HybridCompute"
+
+func resourceServerVulnerabilityAssessment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServerVulnerabilityAssessmentCreate,
+		Read:   resourceServerVulnerabilityAssessmentRead,
+		Delete: resourceServerVulnerabilityAssessmentDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"virtual_machine_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: computeValidate.VirtualMachineID,
+			},
+			"hybrid_machine_id": {
```

Comment:
Is it possible to merge "virtual_machine_id" and "hybrid_machine_id" to one property "target_machine_id"?

Reply:
Yes I had it like that fist. But then got the suggestion to split into separate properties. Do you have any strong arguments to just have one property?
I like the name virtual_machine_id since that is the most commonly used when you reference a VM.

Reply:
I think its better to have it split as it allows for better validation and becomes more obvious, alternatively it might make sense to split this into two resources..